.github/workflows/build_and_deploy.yml

```diff
@@ -90,10 +94,28 @@ jobs:
     strategy:
       matrix: ${{fromJson(needs.fetch.outputs.boards)}}
     steps:
+      - name: Check tests status
+        id: check-tests
+        if: ${{ matrix.board  != null }}
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPO: ${{ github.repository }}
+          COMMIT: ${{ needs.fetch.outputs.merge_commit }}
+        run: |
+          prid=$(gh api -H "Accept: application/vnd.github+json" /repos/$REPO/commits/$COMMIT --jq '.commit.message' | head -n1 | cut -d "#" -f2 | awk '{ print $1}')
+          status_url=$(gh api -H "Accept: application/vnd.github+json" /repos/$REPO/pulls/$prid --jq '._links.statuses.href')
+          testbot_job="testbot-${{ matrix.board }}"
+          passed="no"
+          if curl -sL "${status_url}" | jq -e '.[] | select(.context == "'"${testbot_job}"'") | select(.description == "OS tests have passed\n ")' > /dev/null 2>&1; then
```

Comment:
I guess we need to make sure the string `OS tests have passed` never changes in the Jenkins jobs.